Source/Diffusion/DiffusionSrcForState_N.cpp

```diff
@@ -343,30 +348,42 @@ DiffusionSrcForState_N (const amrex::Box& bx, const amrex::Box& domain, int n_st
     // Using Deardorff
     if (l_use_deardorff && n_end >= RhoKE_comp) {
         int qty_index = RhoKE_comp;
+        amrex::Real l_C_k = solverChoice.Ck;
         amrex::ParallelFor(bx,[=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
         {
-            // Add Buoyancy Source
             Real eps       = std::numeric_limits<Real>::epsilon();
-            Real theta     = cell_prim(i,j,k,PrimTheta_comp);
             Real dtheta_dz = 0.5*(cell_prim(i,j,k+1,PrimTheta_comp)-cell_prim(i,j,k-1,PrimTheta_comp))*dz_inv;
             Real E         = cell_prim(i,j,k,PrimKE_comp);
-            Real denom     = std::abs(grav_gpu[2]) * dtheta_dz / theta;
+            Real strat     = std::abs(grav_gpu[2]) * dtheta_dz * l_inv_theta0; // stratification
             Real length;
-            if (denom <= eps) {
+            if (strat <= eps) {
                 length = l_Delta;
             } else {
-              length = 0.76*std::sqrt(E) / std::sqrt(denom);
+              length = 0.76 * std::sqrt(E / strat);
             }
-            Real KH   = 0.1 * (1.+2.*length/l_Delta) * std::sqrt(E);
-            cell_rhs(i,j,k,qty_index) += cell_data(i,j,k,Rho_comp) * grav_gpu[2] * KH * dtheta_dz;
 
-            // TKE production
+            // From eddy viscosity mu_turb = rho * C_k * l * KE^(1/2), the
+            // eddy diffusivity for heat, KH = (1 + 2*l/delta) * mu_turb
+            Real KH = cell_data(i,j,k,Rho_comp) * l_C_k * length * (1.+2.*length/l_Delta) * std::sqrt(E);
+
+            // Add Buoyancy Source
+            // where the SGS buoyancy flux tau_{theta,i} = -KH * dtheta/dx_i,
+            // such that for dtheta/dz < 0, there is a positive (upward) heat flux;
+            // the TKE buoyancy production is then g/theta_0 * tau_{theta,w}
+            hfx_x(i,j,k) = 0.0;
+            hfx_y(i,j,k) = 0.0;
+            hfx_z(i,j,k) = -KH * dtheta_dz;
```

Comment:
I'm not sure this is the best place to store the heat flux? Specifically, this part of the code is for RhoKE sources and only gets triggered for Deardorff. By contrast, we compute the diffusive fluxes up above and that code is triggered for any turbulence model. If there is an issue with the diffusivity with theta, we may need to modify ComputeTurbulentViscosity.